agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/processors/AgentProcessor.java

```diff
@@ -85,15 +85,13 @@ public boolean isMatch(Attributes attributes, String name) {
     private boolean checkAttributes(Attributes attributes) {
       for (ProcessorAttribute attribute : processorAttributes) {
         // All of these attributes must match exactly for a match to occur.
-        Object existingAttributeValue = attributes.get(AttributeKey.stringKey(attribute.key));
-        // to get the string value
-        // existingAttributeValue.toString()
-        // String.valueOf(existingAttributeValue);
-        if (!(existingAttributeValue instanceof String)) {
+        Object valueObject = attributes.get(attribute.getAttributeKey());
+        if (valueObject == null) {
           // user specified key not found
           return false;
         }
-        if (attribute.value != null && !existingAttributeValue.equals(attribute.value)) {
+        String existingAttributeValue = String.valueOf(valueObject);
+        if (attribute.value != null && !existingAttributeValue.equals(attribute.getStringValue())) {
```

Comment:
for strict can we do equals on object values? (e.g. comparing double object / boolean object)
```suggestion
        if (attribute.value != null && !valueObject.equals(attribute.getValue())) {
```

Reply:
this failed strict type

Reply:
what was the problem? I think it would be good to see what it would require to make it work